docs/source/metrics.rst

```diff
@@ -188,6 +198,63 @@ In practise this means that:
     val = metric(pred, target) # this value can be backpropagated
     val = metric.compute() # this value cannot be backpropagated
 
+****************
+MetricCollection
+****************
+
+In many cases it is beneficial to evaluate the model output by multiple metrics.
+In this case the `MetricCollection` class may come in handy. It accepts a sequence
+of metrics and wraps theses into a single callable metric class, with the same
+interface as any other metric.
+
+Example:
+
+.. code-block:: python
+
+    from pytorch_lightning.metrics import MetricCollection, Accuracy, Precision, Recall
+    target = torch.tensor([0, 2, 0, 2, 0, 1, 0, 2])
+    preds = torch.tensor([2, 1, 2, 0, 1, 2, 2, 2])
+    metric_collection = MetricCollection([Accuracy(),
+                                          Precision(num_classes=3, average='macro'),
+                                          Recall(num_classes=3, average='macro')])
+    metric_collection(preds, target)
+    {'Accuracy': tensor(0.1250),
+     'Precision': tensor(0.0667),
+     'Recall': tensor(0.1111)}
+
+Similarly it can also reduce the amount of code required to log multiple metrics
+inside your LightningModule
+
+.. code-block:: python
+
+    def __init__(self):
+        ...
+        metrics = pl.metrics.MetricCollection(...)
+        self.train_metrics = metrics.clone()
```

Comment:
Can we automate move the clone within the __setattr__ ?

Reply:
I also wouldn't do this, since if you don't have multiple instances (e.g. some metrics you only want to calc for validation or so) you don't need to clone and cloning would only be overhead.